### PR TITLE
dum 0.1.8 (new formula)

### PR DESCRIPTION
**Pull request by lianghx-319**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by lianghx-319** [comment]
https://github.com/egoist/dum/issues/11

File: Formula/dum.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  homepage "https://github.com/egoist/dum"
+  url "https://github.com/egoist/dum/archive/v0.1.8.tar.gz"
+  sha256 "0c8337d79e0e331b2dbd14e615adc9d359d3491632e21d479a306c9a2845009c"
+  license "MIT"
+  head "https://github.com/egoist/dum.git", branch: "master"
+
+  livecheck do
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
```
Comment (carlocab) [inline Formula/dum.rb:12]: Is this needed? i.e. is the default `livecheck` wrong if you omit this?

File: Formula/dum.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  depends_on "rust" => :build
+
+  uses_from_macos "zlib"
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "false"
```
Comment (carlocab) [inline Formula/dum.rb:23]: This needs a proper test.

File: Formula/dum.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class Dum < Formula
+  desc "Npm scripts runner written in Rust"
+  homepage "https://github.com/egoist/dum"
+  url "https://github.com/egoist/dum/archive/v0.1.8.tar.gz"
+  sha256 "0c8337d79e0e331b2dbd14e615adc9d359d3491632e21d479a306c9a2845009c"
+  license "MIT"
+  head "https://github.com/egoist/dum.git", branch: "master"
```
Comment (carlocab) [inline Formula/dum.rb:7]:
```suggestion
  head "https://github.com/egoist/dum.git", branch: "main"
```

File: Formula/dum.rb (hunk 1 above)
Comment (lianghx-319) [reply Formula/dum.rb:12]: I just copy from `fnm.rb`. The default behavior is detecting Github release?

File: Formula/dum.rb (hunk 2 above)
Comment (lianghx-319) [reply Formula/dum.rb:23]: I am finding a test suit to make sure to install it successfully. Should be added tomorrow.

File: Formula/dum.rb (hunk 1 above)
Comment (carlocab) [reply Formula/dum.rb:12]: Try deleting it and then check the output of `brew livecheck dum`. If it's wrong we can keep this. If it's not then we don't need it. Whether you need it or not depends on the repository.

File: Formula/dum.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/dum.rb:23]: Please don't try and run an upstream test suite, those are not the types of tests we're looking for here. See https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/dum.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/dum.rb:12]: Yes, provided the url is a github repo

File: Formula/dum.rb (hunk 2 above)
Comment (lianghx-319) [reply Formula/dum.rb:23]: > Please don't try and run an upstream test suite, those are not the types of tests we're looking for here. See https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Thanks, I will finish in right way👌

File: Formula/dum.rb (hunk 1 above)
Comment (lianghx-319) [reply Formula/dum.rb]: It works!

**Comment by lianghx-319** [comment]
How can I fix this issue?
image

**Comment by SMillerDev** [comment]
Hi, this seems like a nice software, but not notable enough for us to package in Homebrew for now. This is not a decision on the merits of the software, but we cannot distribute every single piece of software and have to draw a line on “notability” somewhere. Of course, we would be happy to reconsider in the future.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Pull request closed without merging**